ompi/mca/osc/ucx/osc_ucx_comm.c

```diff
@@ -582,13 +818,29 @@ int ompi_osc_ucx_compare_and_swap(const void *origin_addr, const void *compare_a
     }
 
     ompi_datatype_type_size(dt, &dt_bytes);
-    ret = opal_common_ucx_wpmem_cmpswp(module->mem,*(uint64_t *)compare_addr,
-                                     *(uint64_t *)origin_addr, target,
-                                     result_addr, dt_bytes, remote_addr);
-    if (ret != OMPI_SUCCESS) {
+    if (sizeof(uint64_t) < dt_bytes) {
+        return OMPI_ERR_NOT_SUPPORTED;
+    }
+
+    uint64_t compare_val;
+    memcpy(&compare_val, compare_addr, dt_bytes);
+    memcpy(result_addr, origin_addr, dt_bytes);
+    ret = opal_common_ucx_wpmem_fetch_nb(module->mem, UCP_ATOMIC_FETCH_OP_CSWAP,
```

Comment:
Could you clarify, why we need to replace `opal_common_ucx_wpmem_cmpswp`?

Reply:
There is no `nb` version of `opal_common_ucx_wpmem_cmpswp`, it calls `opal_common_ucx_atomic_cswap` (and in turn `opal_common_ucx_atomic_fetch`) and eventually waits for the operation to complete.